### PR TITLE
Stateful app version v1 fix

### LINKED ISSUE #220: Missing required field "selector" in io.k8s.api.apps.v1.StatefulSetSpec
Executing:
kubectl apply -f es-master-stateful.yaml

Results in error:
error: error validating "es-master-stateful.yaml": error validating data: [ValidationError(StatefulSet.spec): missing required field "selector" in io.k8s.api.apps.v1.StatefulSetSpec, ValidationError(StatefulSet.spec): missing required field "serviceName" in io.k8s.api.apps.v1.StatefulSetSpec]; if you choose to ignore these errors, turn validation off with --validate=false

### PR DESCRIPTION
**Pull request by Fus1oNn**
Hey Paulo!

Here's another fix, thanks @paulavijit [ it should fix #220 ]

Since app/v1 StatefulSet Controller needs spec.selector to match template.metadata.labels
Ref: https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/

The fix includes headless service referenced in es-data-stateful.yaml as spec.serviceName and selector fixes for master-stateful and data-stateful.

Kind Regards!

### CODE REVIEW

File: stateful/es-data-svc.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
```
Comment (pires) [inline stateful/es-data-svc.yaml]: why is this needed?

Comment (Fus1oNn) [reply stateful/es-data-svc.yaml]: From the link I provided above:
"StatefulSets currently require a Headless Service to be responsible for the network identity of the Pods. You are responsible for creating this Service."

When trying to deploy a statefulset without specifying an existing service name, it returns with an error (the one in the #220 issue)

That's why this fix adds a service for each statefulset and label selectors because:
"In 1.8 and later versions, failing to specify a matching Pod Selector will result in a validation error during StatefulSet creation."
